lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -143,8 +143,9 @@ tokenize(String, Line, Column, Opts) ->
 tokenize(String, Line, Opts) ->
   tokenize(String, Line, 1, Opts).
 
-tokenize([], _Line, _Column, #elixir_tokenizer{terminators=[]}, Tokens) ->
-  {ok, lists:reverse(Tokens)};
+tokenize([], _Line, _Column, #elixir_tokenizer{terminators=[], warnings = Warnings}, Tokens) ->
+  UniqueWarnings = ordsets:to_list(ordsets:from_list(Warnings)),
```

Comment:
There is no reason to use ordsets, lists:reverse/2 is the best way to go here. The warnings should be unique by defiition too. No need to try to unify them.

Finally, I think we can do the following here:

```erl
[elixir_errors:erl_warn(Line, File, Msg) || {Line, File, Msg} <- lists:reverse(Warnings)],
```

By doing so we no longer need to return the warnings and we can remove the call from `prepend_warning`. This change should be enough to fix the original issue!